pkg/types/azure/validation/machinepool.go

```diff
@@ -13,6 +13,8 @@ func ValidateMachinePool(p *azure.MachinePool, poolName string, platform *azure.
 
 	if p.OSDisk.DiskSizeGB < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("diskSizeGB"), p.OSDisk.DiskSizeGB, "Storage DiskSizeGB must be positive"))
+	} else if platform.CloudName == azure.StackCloud && p.OSDisk.DiskSizeGB != 0 && (p.OSDisk.DiskSizeGB < 128 || p.OSDisk.DiskSizeGB > 1023) {
```

Comment:
Can we define these numbers as contants somewhere with a link to the Azure docs where they are specified?